### PR TITLE
drop astring dependency

### PR DESCRIPTION
**Pull request by hannesm**

### CODE REVIEW

**Comment by hannesm** [comment]
this together with mirage-qubes 0.9.5 and xenstore 2.3.0 (both waiting on being merged to opam-repository) reduces the binary size of the firewall to 3.4MB. I'll update this PR once the above packages have been merged to opam-repository, and then update the checksum and call it ready :)

**Comment by palainp** [comment]
Thank you! I propose to update Dockerfile once opam-repository PRs will be merged and merge both PRs into main. And sadly I see I conflicted `build-with.sh` when I removed the space before the hashsum :(

**Comment by hannesm** [comment]
Sounds good to me, I just force-pushed to remove the update of the hash here. Once opam-repository is fine, we can do as you suggested.

**Comment by hannesm** [comment]
Thanks to @dinosaure who merged the outstanding opam-repository PRs, I pushed an update to the opam-repo commit and wait for the checksum to materialize.

**Comment by palainp** [comment]
Thank you that's perfect!

File: build-with.sh [hunk 1]
```diff
@@ -20,5 +20,5 @@ $builder build -t qubes-mirage-firewall .
 echo Building Firewall...
 $builder run --rm -i -v `pwd`:/tmp/orb-build:Z qubes-mirage-firewall
 echo "SHA2 of build:   $(sha256sum ./dist/qubes-firewall.xen)"
-echo "SHA2 last known: 163991ea96842e03d378501a0be99057ad2489440aff8ae81d850624d98fd3f0"
+echo "SHA2 last known: 6225a8a5e8dbaa9a39619770ec730c4db661625d9aaee407b571c0e786cc3a81"
```
Comment (palainp) [inline build-with.sh:23]: I'm unable to check from my smartphone, #195 needs only 1 space between ':' and the hashsum. If that's the case LGTM :)

**Review by palainp — APPROVED** [review]

File: build-with.sh (hunk 1 above)
Comment (palainp) [reply build-with.sh:23]: Ok I can see only 1 now, so it's fine to me!

**Comment by hannesm** [comment]
and CI is green, so feel free to cut a release when you feel like doing so :)

**Comment by palainp** [comment]
I'll run a couple of local tests and cut the new release. Thank you @hannesm

**Comment by palainp** [comment]
Appart from that issue (locally fixed by hand), all went well. So once the new release is merged in opam-repository, it should be ok to cut the release :)

**Comment by hannesm** [comment]
waiting for https://github.com/ocaml/opam-repository/pull/25823

**Comment by hannesm** [comment]
Dear @palainp, this is now updated to mirage-net-xen 2.1.5. Please feel free to test and merge and release :)

**Comment by palainp** [comment]
Thank you, all is fine!

**Pull request merged**